src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -226,11 +226,37 @@ private static int GetMaxTaskCount(RestoreArgs restoreContext)
             return maxTasks;
         }
 
-        private static async Task<RestoreSummary> ExecuteAndCommitAsync(RestoreSummaryRequest summaryRequest, CancellationToken token)
+        private static async Task<RestoreSummary> ExecuteAndCommitAsync(RestoreSummaryRequest summaryRequest, IRestoreProgressReporter progressReporter, CancellationToken token)
         {
-            var result = await ExecuteAsync(summaryRequest, token);
+            RestoreResultPair result = await ExecuteAsync(summaryRequest, token);
+            bool isNoOp = result.Result is NoOpRestoreResult;
+            IReadOnlyList<string> filesToBeUpdated = isNoOp ? null : GetFilesToBeUpdated(result);
+
+            if (!isNoOp)
+            {
+                progressReporter?.StartProjectUpdate(summaryRequest.Request.Project.FilePath, filesToBeUpdated);
+            }
+
+            RestoreSummary summary = await CommitAsync(result, token);
 
-            return await CommitAsync(result, token);
+            if (!isNoOp)
+            {
+                progressReporter?.EndProjectUpdate(summaryRequest.Request.Project.FilePath, filesToBeUpdated);
```

Comment:
Doesn't doing this in `ExecuteAndCommitAsync` instead of `CommitAsync` mean that these events will not fire when a package is installed in PMUI, since it does the execute, show the preview window and then commit only if the customer accepts the preview changes?

Reply:
I didn't read the PR comment. This is a known issue that will be followed up on.